client/gutenberg/extensions/slideshow/edit.js

```diff
@@ -0,0 +1,181 @@
+			const { columns } = this.props.attributes;
+			this.setState( { selectedImage: null } );
+			this.props.setAttributes( {
+				images,
+				columns: columns ? Math.min( images.length, columns ) : columns,
+			} );
+		};
+	};
+	updateAlignment = value => {
+		this.props.setAttributes( { align: value } );
+	};
+	addFiles( files ) {
+		const currentImages = this.props.attributes.images || [];
+		const { noticeOperations, setAttributes } = this.props;
+		mediaUpload( {
+			allowedTypes: ALLOWED_MEDIA_TYPES,
+			filesList: files,
+			onFileChange: images => {
+				const imagesNormalized = images.map( image => pickRelevantMediaFiles( image ) );
+				setAttributes( {
+					images: currentImages.concat( imagesNormalized ),
+				} );
+			},
+			onError: noticeOperations.createErrorNotice,
+		} );
+	}
+	render() {
+		const { attributes, className, noticeOperations, noticeUI, setAttributes } = this.props;
+		const { align, effect, images } = attributes;
+		const controls = (
+			<Fragment>
+				<InspectorControls>
+					<SelectControl
+						label={ __( 'Transition effect' ) }
+						value={ effect }
+						onChange={ value => {
+							setAttributes( { effect: value } );
+						} }
+						options={ settings.effectOptions }
+					/>
+				</InspectorControls>
+				<BlockControls>
+					<BlockAlignmentToolbar
+						value={ align }
+						onChange={ this.updateAlignment }
+						controls={ [ 'center', 'wide', 'full' ] }
+					/>
+					{ !! images.length && (
+						<Toolbar>
+							<MediaUpload
+								onSelect={ this.onSelectImages }
+								allowedTypes={ ALLOWED_MEDIA_TYPES }
+								multiple
+								gallery
+								value={ images.map( img => img.id ) }
+								render={ ( { open } ) => (
+									<IconButton
+										className="components-toolbar__control"
+										label={ __( 'Edit Slideshow' ) }
+										icon="edit"
+										onClick={ open }
+									/>
+								) }
+							/>
+						</Toolbar>
+					) }
+				</BlockControls>
+			</Fragment>
+		);
+
+		if ( images.length === 0 ) {
+			return (
+				<Fragment>
+					{ controls }
+					<MediaPlaceholder
+						icon="format-gallery"
+						className={ className }
+						labels={ {
+							title: __( 'Slideshow' ),
+							instructions: __( 'Drag images, upload new ones or select files from your library.' ),
+						} }
+						onSelect={ this.onSelectImages }
+						accept="image/*"
+						allowedTypes={ ALLOWED_MEDIA_TYPES }
+						multiple
+						notices={ noticeUI }
+						onError={ noticeOperations.createErrorNotice }
+					/>
+				</Fragment>
+			);
+		}
+		return (
+			<Fragment>
+				{ controls }
+				{ noticeUI }
+				<Slideshow className={ className } effect={ effect } align={ align }>
+					{ images.map( ( image, index ) => {
+						const { alt, caption, height, id, url, width } = image;
+						return (
+							<div className="wp-block-slideshow_image_container" key={ index }>
+								<img
+									src={ url }
+									alt={ alt }
+									data-is-image={ true }
```

Comment:
This should be synonymous to:

```suggestion
									data-is-image
```

It's a bit odd to me that we're setting a data-attribute to a JavaScript value. Is this intended to be serialized and read from HTML? In that case, we might prefer the string:

```suggestion
									data-is-image="true"
```

Reply:
Confirmed... you're right. Fixed in https://github.com/Automattic/wp-calypso/pull/29419/commits/56874762f99209aacad76d604fdc58067be2466b